rust-runtime/aws-smithy-runtime/src/client/retries/strategy/standard.rs

```diff
@@ -84,6 +93,61 @@ impl StandardRetryStrategy {
             p.forget()
         }
     }
+
+    fn calculate_backoff(
+        &self,
+        cfg: &ConfigBag,
+        retry_reason: Option<&RetryReason>,
+    ) -> Result<Duration, ShouldAttempt> {
+        let request_attempts = cfg
+            .load::<RequestAttempts>()
+            .expect("at least one request attempt is made before any retry is attempted")
+            .attempts();
+        let token_bucket = cfg.load::<TokenBucket>();
+
+        match retry_reason {
+            Some(RetryReason::Explicit(backoff)) => Ok(*backoff),
+            Some(RetryReason::Error(kind)) => {
+                update_rate_limiter_if_exists(cfg, kind == &ErrorKind::ThrottlingError);
+                if let Some(delay) = check_rate_limiter_for_delay(cfg, *kind) {
```

Comment:
You should be able to get rid of the reference annoyance:
```suggestion
            Some(RetryReason::Error(&kind)) => {
                update_rate_limiter_if_exists(cfg, kind == ErrorKind::ThrottlingError);
                if let Some(delay) = check_rate_limiter_for_delay(cfg, kind) {
```